mdformat/plugins.py

```diff
@@ -15,3 +20,33 @@ def _load_codeformatters() -> Dict[str, Callable[[str, str], str]]:
 
 
 CODEFORMATTERS: Mapping[str, Callable[[str, str], str]] = _load_codeformatters()
+
+
+class ParsePluginAbstract:
+    """A class for extending the base parser."""
+
+    @staticmethod
+    def update_mdit(mdit: MarkdownIt) -> MarkdownIt:
+        """Update the parser, e.g. by adding a plugin: `mdit.use(myplugin)`"""
+        return mdit
+
+    @staticmethod
+    def render_token(
+        renderer: MDRenderer, tokens: List[Token], index: int, options: dict, env: dict
+    ) -> Optional[Tuple[str, int]]:
+        """Convert a token to a string, or return None if no render method
+        available.
+
+        :returns: (text, index) where index is of the final "consumed" token
+        """
+        return None
+
+
+def _load_extendplugins() -> Mapping[str, ParsePluginAbstract]:
+    extendplugins_entrypoints = importlib_metadata.entry_points().get(
+        "mdformat.extendplugins", ()
```

Comment:
The group for codeformatters was made singular form `mdformat.codeformatter`, so maybe this too for consistency? Not sure if one or the other is more appropriate.

Reply:
On second though, would something like `"mdformat.parser_extension`" be more descriptive? Maybe the constant below should also be `mdformat.plugins.PARSER_EXTENSIONS`. The plugin part should be implied from the namespace, i think.

Reply:
Oh and if we do this name change, it should probably be applied to keyword arguments of `mdformat.file()` and `mdformat.text()` as well

Reply:
yep sounds reasonable